### PR TITLE
NOJIRA: Add Ansible docs

### PR DESCRIPTION
**Pull request by gregorydlogan**
This documentation adds links to the Ansible installation scripts to our mainline documentation.  Aiming at 4.x currently with the idea that it would percolate upwards from there.

### CODE REVIEW

**Comment by gregorydlogan** [comment]
Note that I've recently identified an issue with the scripts and our 4.x+ codebase.  If you're testing this, please use the 3.x branch in the script repo until I post here otherwise.

**Comment by gregorydlogan** [comment]
4.x and newer should work now!

File: docs/guides/admin/docs/installation/ansible.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+Install via Script using the Repository (Debian, Ubuntu)
```
Comment (lkiesow) [inline docs/guides/admin/docs/installation/ansible.md]: Maybe `Install via Ansible using the Repository (Debian, Ubuntu)` instead?

Comment (lkiesow) [reply docs/guides/admin/docs/installation/ansible.md]: > The Ansible scripts are tested on […] CentOS 7+

But the title says they are only for Debian and Ubuntu. Remove the distros from the title?

File: docs/guides/admin/docs/installation/ansible.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+===========================================================================
+
+Installing Opencast can be complex, so we have built a set of Ansible scripts to handle the installation and 
+configuration.  These scripts should handle most common installation setups, and are ideal for someone wishing to 
+quickly and easily set up Opencast for testing.  These scripts are also a good base upon which to build your own, local
+configuration management scripts.
+
+The Ansible scripts are tested on Debian 8+, Ubuntu 16.04+, and CentOS 7+, but may also work on other systems.
```
Comment (lkiesow) [inline docs/guides/admin/docs/installation/ansible.md]: What do you mean by this?

1. The ￼Ansible scripts are tested on Debian 8, Ubuntu 16.04, and CentOS 7, but may also work on other systems.
2. The ￼Ansible scripts should work, starting with Debian 8, Ubuntu 16.04, and CentOS 7, but may also work on other systems.

E.g. the `CentOS 7+` implies that you've tested the scripts on CentOS 8 already.

File: docs/guides/admin/docs/installation/ansible.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+get by mail after the registration to successfully complete this manual. The placeholders `[your_username]` and 
+`[your_password]` are used in this manual wherever the credentials are needed.
+
+[Please visit https://pkg.opencast.org](https://pkg.opencast.org)
+
+
+Testing Use
+-----------
+
+The scripts can be found [here](https://github.com/gregorydlogan/oc-config-management).  If you are familiar with `git`
```
Comment (lkiesow) [inline docs/guides/admin/docs/installation/ansible.md]: When we list them in the official docs, does it make sense to move them into the Opencast organization?

File: docs/guides/admin/docs/installation/ansible.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+make a local clone, otherwise download a zip file of the scripts.  Read through the documentation, which contains
+additional setup steps.
+
+Once you have set up your host configuration in `hosts`, run 
+`ansible-playbook -K -i hosts opencast.yml --extra-vars "repo_username=[your_username] repo_password=[your_password]"`.
+If you are uncomfortable with your username and password on the commandline, you can set those variables in 
+`group_vars/all.yml`.
```
Comment (lkiesow) [inline docs/guides/admin/docs/installation/ansible.md]: Maybe use [Ansible Vault](https://docs.ansible.com/ansible/2.4/vault.html)?

File: docs/guides/admin/docs/installation/ansible.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+advanced option.  Read the script documentation for more detail.
+
+Production Use
+--------------
+
+Similar to testing, you need to read the documentation and perform the additional installation steps.  The only
+differences between testing and production use are:
+
+ - Setting different passwords for all password options in `group_vars/all.yml`.  These scripts default to the common
+   passwords used worldwide by Opencast.  You do not want to use these passwords in production.
```
Comment (lkiesow) [inline docs/guides/admin/docs/installation/ansible.md]: I hope they are not used worldwide. At least not in production :-D
Maybe `used by default` instead?

Comment (lkiesow) [reply docs/guides/admin/docs/installation/ansible.md]: And again, ansible vault may be a good idea.

File: docs/guides/admin/docs/installation/ansible.md (hunk 2 above)
Comment (gregorydlogan) [reply docs/guides/admin/docs/installation/ansible.md]: Ah, good point.  The intention was to highlight that these were tested on Debians 8, 9, Ubuntus 16.04, 18.04, and Centos 7.  As stable releases are released (Deb 10, Ubu 20.04, CentOS 8) I'll test against those.

Something like this?
`
The Ansible scripts are tested on stable releases of Debian, Ubuntu, and CentOS, currently Debian 8, and 9, Ubuntu 16.04, and 18.04, and CentOS 7.
`

File: docs/guides/admin/docs/installation/ansible.md (hunk 3 above)
Comment (gregorydlogan) [reply docs/guides/admin/docs/installation/ansible.md]: I'm fine with that.

File: docs/guides/admin/docs/installation/ansible.md (hunk 4 above)
Comment (gregorydlogan) [reply docs/guides/admin/docs/installation/ansible.md]: I was going with absolute simplicity here, but you're probably right that there should be more work on this section.  I'll add a note that there are better ways to do this.

File: docs/guides/admin/docs/installation/ansible.md (hunk 5 above)
Comment (gregorydlogan) [reply docs/guides/admin/docs/installation/ansible.md]: I've caught multiple users running with web-accessible systems and default passwords :)

**Comment by gregorydlogan** [comment]
This should be ready for further review now.

**Comment by gregorydlogan** [comment]
Adding on hold since I've identified a few more issues in the scripts and want to rebase them out of the public tree prior to releasing this.

**Comment by KatrinIhler** [comment]
@gregorydlogan I guess this is still on hold?

**Comment by gregorydlogan** [comment]
For the moment yes, although I'm planning on fixing this by EOW now that I'm back online.

**Comment by gregorydlogan** [comment]
This should be ready to go.  Once these go in we need to fork the ansible scripts into the opencast project namespace as well.

**Comment by JamesUoM** [comment]
> This should be ready to go. Once these go in we need to fork the ansible scripts into the opencast project namespace as well.

Does this need to go in against r/5.x ?

**Comment by gregorydlogan** [comment]
The intention at the time was to merge it forward from 4.x  I'm fine pointing this at 5.x - the scripts have their own branches so it really doesn't matter where they enter the Opencast codebase :)

File: docs/guides/admin/docs/installation/ansible.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+
+Setup
+-----
+
+These scripts use [Ansible](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html), which
+allows you to easily and automatically manage large numbers of machines via SSH.  Install Ansible before continuing
+with this guide.
+
+The scripts can be found [here](https://github.com/opencast/oc-config-management).  If you are familiar with `git`
```
Comment (JamesUoM) [inline docs/guides/admin/docs/installation/ansible.md:20]: If we merge this now this link is broken.

Comment (gregorydlogan) [reply docs/guides/admin/docs/installation/ansible.md:20]: The intention is to fork the code at https://github.com/gregorydlogan/oc-config-management at merge time :)

Comment (JamesUoM) [reply docs/guides/admin/docs/installation/ansible.md:20]: I was thinking this is the first ever case of the documentation been present before the code!

**Comment by gregorydlogan** [comment]
Creating the upstream fork now.

**Pull request merged**